subsys/bluetooth/mesh/sensor_cli.c

```diff
@@ -573,13 +573,16 @@ int bt_mesh_sensor_cli_desc_all_get(struct bt_mesh_sensor_cli *cli,
 
 	struct list_rsp list_rsp = {
 		.sensors = sensors,
-		.count = *count,
+		.count = count ? *count : 0,
```

Comment:
Could arguably be `(sensors && count) ? *count : 0` in all instances. We never check `sensors != NULL` in `handle_status` once we get past `count > 0`

Reply:
If either sensor or count is not NULL, the other argument shall not be NULL since both are used in pair. I though that this is kind of an agreement in API we have where it is user's responsibility to pass all arguments correctly. We don't pass ack ctx to `model_ackd_send` if `sensors` is NULL, thus, in `handle_status` and `handle_descriptor_status` we won't retrieve any ack ctx when calling `bt_mesh_msg_ack_ctx_match`.

Reply:
That's fair